### ISSUE TITLE
ActionDispatch::Routing::UrlFor does not define default_url_options reader when included into a module

### ISSUE THREAD

**Issue by dreyks** [rails/rails#24968]
When including `ActionDispatch::Routing::UrlFor` into a module that then gets included into a class, the `default_url_options` reader is missing, only writer is present. When including `ActionDispatch::Routing::UrlFor` directly into any class, everything is ok

https://github.com/rails/rails/blob/5-0-stable/actionpack/lib/action_dispatch/routing/url_for.rb#L95
### Steps to reproduce

``` ruby
module ProxyModule
   include ActionDispatch::Routing::UrlFor
 end

class ClassWithProxy
   include ProxyModule
end

class DirectClass
  include ActionDispatch::Routing::UrlFor
end

DirectClass.new.url_options
ClassWithProxy.new.url_options

```
### Expected behavior

```
{}
{}
```
### Actual behavior

```
{}
NameError: undefined local variable or method `default_url_options' for #<SomeClass:0x007f30d629a2d0 # @_routes=nil>
Did you mean?  default_url_options=
               @@default_url_options
from /bundle/gems/actionpack-5.0.0.rc1/lib/action_dispatch/routing/url_for.rb:113:in `url_options'
```
### System configuration

**Rails version**: 5.0.0.rc1

**Ruby version**: 2.3.1

**Comment by maclover7** [comment]
This is the expected behavior -- `default_url_options` is a `class_attribute`, which means it cannot be accessed on instances of classes, only on the class itself.

With your example above with an instance of  `SomeClass`, you would have to call `self.class.default_url_options`, instead of `self.default_url_options`.

**Comment by dreyks** [comment]
Still
```ruby
DirectClass.default_url_options # => {}
ClassWithProxy.default_url_options # => undefined method `default_url_options'
```

It's either I get it wrong, or I didn't explain it clear

**Comment by maclover7** [comment]
Yes -- I think this is a limitation of the implementation here. If you want to have a reader, you will need to include it in a class -- not a module. The reader is only created when it is included into a class.

Let's take apart the `included` block:
```ruby
        unless method_defined?(:default_url_options)
          # This will return false, because you are including into a module
          if respond_to?(:class_attribute)
           # A reader is created here... (class!)
            class_attribute :default_url_options
          else
           # But not here... (module!)
            mattr_writer :default_url_options
          end

          self.default_url_options = {}
        end
```

As I said, I think this is more of hitting on the limits of the implementation, but is not a bug 😬

**Comment by dreyks** [comment]
Why not `mattr_accessor` ? =)

Anyway this popped up when migrating to Rails5 and it only happens in RSpec specs. The included hook has not been changed since 4.2-stable, so it's something inside RSpec. (Probably they mix url_for in another way now)

**Comment by simpl1g** [comment]
Same for me in helper specs in rspec, trying to migrate to Rails 5.0.0.rc2 with rspec master
